eng/Version-Increment.ps1

```diff
@@ -17,85 +17,128 @@ The Name of the Package
 $CsprojData = New-Object -TypeName XML
 $PackageCsprojPath = Join-Path $RepoRoot "sdk" $ServiceDirectory $PackageDirName "src" "${PackageName}.csproj"
 $ChangelogPath = Join-Path $RepoRoot "sdk" $ServiceDirectory $PackageDirName "CHANGELOG.md"
 $CsprojData.Load($PackageCsprojPath)
-
 $PackageVersion = Select-XML -Xml $CsprojData -XPath '/Project/PropertyGroup/Version'
-if ([System.String]::IsNullOrEmpty($PackageVersion))
-{
-    Write-Error "Missing Version property in project file ${PackageCsprojPath}"
-    exit 1
-}
 
-Write-Verbose "Current Version: ${PackageVersion}"
 
-$Major = $Minor = $Patch = $Pre = $PreTag = $PreVer = $Null
-
-if ($PackageVersion -Match $VERSION_REGEX)
-{
-    if($Matches[4] -eq $null){$Pre = @()}
-    else{$Pre = $Matches[4].Split(".")}
-
-    $Major = [int]$Matches[1]
-    $Minor = [int]$Matches[2]
-    $Patch = [int]$Matches[3]
-    $PreTag = [string]$Pre[0]
-    $PreVer = [int]$Pre[1]
-}
-else
-{
-    Write-Error "Version property contains incorrect format. It should be in format 'X.Y.Z[-preview.N]'"
-    exit 1
-}
-
-
-# Increment Version
-if ([System.String]::IsNullOrEmpty($PreTag))
-{
-    $PreTag = 'preview'
-    $PreVer = 1
-    $Minor++
-    $Patch = 0
-}
-elseif ($PreTag -eq 'preview' -and $PreVer -ne $Null)
+if ([System.String]::IsNullOrEmpty($NewVersionString))
 {
-    $PreVer++
+    $VersionTable = Parse-Version($PackageVersion)
+    Write-Verbose "Current Version: ${PackageVersion}"
+    # Increment Version
+    if ([System.String]::IsNullOrEmpty($VersionTable['pretag']))
+    {
+        $VersionTable['pretag'] = 'preview'
+        $VersionTable['prever'] = 1
+        $VersionTable['minor']++
+        $VersionTable['patch'] = 0
+    }
+    else
+    {
+        $VersionTable['prever']++
+    }
+
+    $NewPackageVersion = "{0}.{1}.{2}-{3}.{4}" -F $VersionTable['major'], $VersionTable['minor'], $VersionTable['patch'], $VersionTable['pretag'], $VersionTable['prever']
+    Update-Version
 }
 else
 {
-    Write-Error "Unexpected pre-release identifier '${PreTag}'"
-    exit 1
-}
-
-
-$NewPackageVersion = "{0}.{1}.{2}-{3}.{4}" -F $Major, $Minor, $Patch, $PreTag, $PreVer
-Write-Verbose "New Version: ${NewPackageVersion}"
-${PackageVersion}.Node.InnerText = $NewPackageVersion
-$CsprojData.Save($PackageCsprojPath)
-
-# Increment Version in ChangeLog file
-& "${PSScriptRoot}/common/Update-Change-Log.ps1" -Version $NewPackageVersion -ChangeLogPath $ChangelogPath
\ No newline at end of file
+    # Use specified VersionString
+    $VersionTable = Parse-Version($NewVersionString)
+    if ($VersionTable['pretag'] -eq '')
+    {
+        $NewPackageVersion = "{0}.{1}.{2}" -F $VersionTable['major'], $VersionTable['minor'], $VersionTable['patch']
+    }
+    else
+    {
+        $NewPackageVersion = "{0}.{1}.{2}-{3}.{4}" -F $VersionTable['major'], $VersionTable['minor'], $VersionTable['patch'], $VersionTable['pretag'], $VersionTable['prever']
+    }
+    Update-Version -ReplaceVersion $True
```

Comment:
In this case I suspect most people also want to change from unreleased to stamp a date, as they will generally call this right before they release to set the final version.

Reply:
Updated the call to `Update-Changelog.ps1` script to handles this